lib/std/fmt.zig

```diff
@@ -1915,6 +1915,54 @@ test "parseUnsigned" {
     try std.testing.expectError(error.InvalidCharacter, parseUnsigned(u8, "", 10));
 }
 
+/// Parses a number like '2G', '2Gi', or '2GiB'.
+pub fn parseIntSizeSuffix(buf: []const u8, radix: u8) ParseIntError!usize {
+    var without_b = buf;
+    if (mem.endsWith(u8, buf, "B")) without_b.len -= 1;
+    var without_i = without_b;
+    var base: usize = 1000;
+    if (mem.endsWith(u8, without_b, "i")) {
+        without_i.len -= 1;
+        base = 1024;
+    }
+    if (without_i.len == 0) return error.InvalidCharacter;
+    const orders_of_magnitude: usize = switch (without_i[without_i.len - 1]) {
+        'k', 'K' => 1,
+        'M' => 2,
+        'G' => 3,
+        'T' => 4,
+        'P' => 5,
+        'E' => 6,
+        'Z' => 7,
+        'Y' => 8,
+        else => 0,
+    };
+    var without_suffix = without_i;
+    if (orders_of_magnitude > 0) {
+        without_suffix.len -= 1;
+    } else if (without_i.len != without_b.len) {
+        return error.InvalidCharacter;
```

Comment:
Can you update the description for ParseIntError? It says `    /// The input was empty or had a byte that was not a digit`.

Reply:
Good idea.